Src/VimCore/Modes_Insert_InsertMode.fs

```diff
@@ -511,14 +512,28 @@ type internal InsertMode
                     let keyInputSet = KeyInputSet(keyInput)
                     RawInsertCommand.InsertCommand (keyInputSet, command, commandFlags) |> Some
 
-                if keyInput.KeyModifiers <> VimKeyModifiers.None && not (CharUtil.IsLetterOrDigit c) then
-                    // Certain keys such as Delete, Esc, etc ... have the same behavior when invoked
-                    // with or without any modifiers.  The modifiers must be considered because they
-                    // do participate in key mapping.  Once we get here though we must discard them
+                if keyInput.KeyModifiers = VimKeyModifiers.Control then
+
+                    // A key with only the control modifier that isn't mapped
+                    // is never a direct insert. But on some international
+                    // keyboards, it might translate to an ASCII control
+                    // character. See issue #2462.
+                    match Map.tryFind keyInput _commandMap with

```

Comment:
When will this `match` ever succeed? It looks like a duplicate of the first `match` call in this method.

Reply:
Ah, right. I cribbed that code as a simplification of the code directly after this section, but that other code is actually checking a different key which isn't redundant.